lib/net.js

```diff
@@ -1258,6 +1258,8 @@ Server.prototype._listen2 = function(address, port, addressType, backlog, fd) {
   }
 
   // generate connection key, this should be unique to the connection
+  if (+port === 0)
+    port = this.address().port;
```

Comment:
Perhaps it would be better to unconditionally create `_connectionKey` from the values in `this.address()`.